emscripten.py

```diff
@@ -1691,27 +1691,23 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
     runtime_assertions = ''
   else:
     runtime_assertions = RUNTIME_ASSERTIONS
-    # assert on the runtime being in a valid state when calling into compiled code. The only exceptions are
-    # some support code
-    receiving_functions = [f for f in exported_implemented_functions if f not in ('_memcpy', '_memset', '_emscripten_replace_memory', '__start_module')]
+    # assert on the runtime being in a valid state when calling into compiled code. The only exceptions are some support code.
+    # WASM=1 already inserts runtime assertions, so no need to do it again here (see create_receiving_wasm)
+    if not shared.Settings.WASM:
```

Comment:
My understanding is that `create_receiving_wasm` is for the wasm backend and `create_receiving` is for fastcomp.   So when running with WASM=1 and fastcomp we still want this code don't we?

Reply:
I'm not sure. That's why I didn't want to restructure the whole thing. What I know is that in `WASM=0` runtime assertions are added once, and in `WASM=1` they are added twice.
I assumed that `create_receiving_wasm` was responsible for that, but I could be wrong.

Reply:
There should be tests for it, but it's possible they are disabled on the wasm backend atm - so we should confirm this before landing.

Reply:
From the code seems clear that either `create_receiving_wasm` *or* `create_receiving` is called, never both.  So at the very least this comment it wrong.  Probably worth investigating..